runtime/src/qio/sys.c

```diff
@@ -1865,45 +1870,53 @@ err_t sys_socketpair(int domain, int type, int protocol, fd_t* sockfd_out_a, fd_
 extern void chpl_task_yield(void);
 
 err_t sys_select(int nfds, fd_set* readfds, fd_set* writefds, fd_set* exceptfds, struct timeval* timeout, int* nset) {
-
   int got_nset;
   err_t err_out = 0;
-  struct timeval deadline;
-  struct timeval now;
-  struct timeval real_timeout;
-  long wait_usec = 5;
-
-  if (timeout != NULL) {
-    gettimeofday(&deadline, NULL);
-    deadline.tv_sec += timeout->tv_sec;
-    deadline.tv_usec += timeout->tv_usec;
-    if (deadline.tv_usec > 1000000) {
-      deadline.tv_sec++;
-      deadline.tv_usec -= 1000000;
+
+  struct timeval temp_timeout;
+  temp_timeout.tv_sec = 0;
+  // 64 milliseconds allows instantaneous connections whereas 32000 doesn't
+  // and is small enough to not block process for longtime
+  temp_timeout.tv_usec = 64000;
+
+  // calculate available timeout
+  if (timeout != NULL){
+    temp_timeout.tv_usec = timeout->tv_usec > 64000 || timeout->tv_sec >= 1 ? 64000 : timeout->tv_usec;
+
+    timeout->tv_usec -= temp_timeout.tv_usec;
```

Comment:
I don't think this code should be modifying timeout. But, you can make a second local copy. Maybe we can have `struct timeval first_timeout;` and `struct timeval second_timeout;` ?

Reply:
made the change